dotnet/CoreLib/Handlers/GenerateEmbeddingsHandler.cs

```diff
@@ -60,22 +60,25 @@ public class GenerateEmbeddingsHandler : IPipelineStepHandler
         {
             // Track new files being generated (cannot edit originalFile.GeneratedFiles while looping it)
             Dictionary<string, DataPipeline.GeneratedFileDetails> newFiles = new();
+            var throttler = new SemaphoreSlim(initialCount: 4);
```

Comment:
nit: A more sensible default may be `Environment.ProcessorCount` so it scales according to the compute offered by the machine.

Reply:
It can be not expected, if for example you have a server with 64 processors, and you have 100 files parallel jobs for process files, then maybe 6400 requests in ada will start to fail on TooManyRequests. I think to make it configurable, what do you think?

Reply:
Is it good idea add int DataPipeline propery for number of parallel tasks? so we can control it

Reply:
Isn't the work IO bound, in which case the number of processors doesn't really matter?

Reply:
if we will have a lot of tasks they will case Thread starvation, 
anyway most importat there is 
``` cs
 (string summary, bool success) = await this.SummarizeAsync(content).ConfigureAwait(false);
```

Reply:
we have to check it wih many files wich prosessed at the same time and I added Environment.ProcessorCount

Reply:
> It can be not expected, if for example you have a server with 64 processors, and you have 100 files parallel jobs for process files, then maybe 6400 requests in ada and it will start to fail on TooManyRequests. I think to make it configurable, what do you think?

IMO configurable is ideal, but I'd argue the resilience should be handled by the underlying provider and not the handler itself.